roles/haproxy/templates/haproxy_frontend.cfg.j2

```diff
@@ -49,10 +52,17 @@ frontend local_ip
     option httplog
     capture request header User-agent len 256
     capture request header X-TLS-Client len 256
-    # The following lines are to log the request rate per 10s
-    stick-table type ip size 1m expire 10s store http_req_rate(10s)
-    http-request track-sc0 src
+    # Allowlist ACL
+    acl allowlist src -f /etc/haproxy/acls/allowedips.acl
+    # Measure and log the request rate per 10s
+    http-request track-sc0 src table st_httpreqs_per_ip
     http-request capture sc_http_req_rate(0) len 4
+    # Create an ACL when the request rate exceeds {{ haproxy_max_request_rate }} per 10s and deny them
+    acl exceeds_max_request_rate_per_ip sc_http_req_rate(0) gt {{ haproxy_max_request_rate }}
+    http-request deny deny_status 429 if exceeds_max_request_rate_per_ip !allowlist
+    # Measure and log the request rate per path and ip 
+    http-request track-sc1 base32+src table st_httpreqs_per_ip_and_path
+    http-request capture sc_http_req_rate(1) len 4
     # Deny the request when no valid host header is used
     http-request deny if ! valid_vhost
```

Comment:
Is there a rationale for the ordering of the checks? I would do them from simplest to less simple: check if method is even allowed, check if hostname even exists (basic sanity - we can reject those out of hand anyway. Only then check if there’s too many of them. It also makes the generated error more specific.

Reply:
No second thought on the ordering was given. It makes more sense indeed to do it in the proposed order. I've changed it